extensions/amp-autocomplete/0.1/test/test-amp-autocomplete.js

```diff
@@ -229,45 +241,79 @@ describes.realWin(
               ['apple'],
               impl.container_
             );
-            expect(impl.container_.children.length).to.equal(1);
+            expect(impl.container_.children).to.have.length(1);
             expect(impl.container_.children[0].innerText).to.equal('apple');
           });
       });
     });
 
-    it('renderResults_() should update the container_ with plain text', () => {
-      const createSpy = env.sandbox.spy(impl, 'createElementFromItem_');
-      return impl.renderResults_(['apple'], impl.container_).then(() => {
-        expect(impl.container_.children.length).to.equal(1);
+    describe('getRemoteData_()', () => {
+      it('should proxy XHR to viewer', async () => {
+        impl.isSsr_ = true;
```

Comment:
Try not to reference ivars in tests if possible.  It reduces refactorability. 

E.g. you can add a test-only `getSsrTemplateHelper()` and stub `isSupported()`.

Reply:
To clarify, this means keeping the public surface clear and not messing with private props.

Concretely, stubbing `isSupported()` would get rid of the need to explicitly override `isSsr_` vs. simply short-circuiting.

`getSsrTemplateHelper()` would pass `impl.ssrTemplateHelper_` through and then you'd just stub the method's result to be static in tests. This one technically has the same internal modification properties but the stubbing flow is clearer since a public method signature should be expected to be stable or referenced elsewhere.